### ISSUE TITLE
Add a `Replay` interface that specifies replay policy

### ISSUE THREAD

**Issue by alexandrepiveteau** [markdown-party/kotlin-echo#68]
A special `Replay` interface could be used to indicate how much replay is required for a specific event insertion. It could look like this :

```kotlin
interface Replay<Event, Model, Change> {
  fun shouldRevert(current: EventLog<Event, Change>, model: Model, event: Event, change: Change): Boolean
}
```

Additionally, some default `Replay` implementations could :

1. Revert operations to follow event identifier ordering.
2. Not revert operations at all (and allow commutative insertions).

**Comment by alexandrepiveteau** [comment]
This is somewhat related to #57.

**Comment by alexandrepiveteau** [comment]
Similarly to #57, this is fixed in #83.
